step-templates/azure-function-deployment.json

```diff
@@ -0,0 +1,85 @@
+  "ActionType": "Octopus.TentaclePackage",
+  "Version": 1,
+  "CommunityActionTemplateId": null,
+  "Properties": {
+    "Octopus.Action.Package.AutomaticallyRunConfigurationTransformationFiles": "True",
+    "Octopus.Action.Package.AutomaticallyUpdateAppSettingsAndConnectionStrings": "True",
+    "Octopus.Action.EnabledFeatures": "Octopus.Features.CustomScripts,Octopus.Features.JsonConfigurationVariables",
+    "Octopus.Action.Package.DownloadOnTentacle": "False",
+    "Octopus.Action.Package.FeedId": "feeds-builtin",
+    "Octopus.Action.Package.JsonConfigurationVariablesEnabled": "True",
+    "Octopus.Action.CustomScripts.PostDeploy.ps1": "$installationPath = $OctopusParameters[\"Octopus.Action.Package.InstallationDirectoryPath\"]\r\n$packageId = $OctopusParameters[\"Octopus.Action.Package.PackageId\"]\r\n$packageVersion = $OctopusParameters[\"Octopus.Action.Package.PackageVersion\"]\r\n\r\nWrite-Host \"Installation Path: $($installationPath)\"\r\nWrite-Host \"Package ID: $($packageId)\"\r\nWrite-Host \"Package Version: $($packageVersion)\"\r\n\r\n$zipFilePath = \"$($installationPath)\\$($packageId).$($packageVersion).zip\"\r\n\r\nWrite-Host \"Zip File Path: $($zipFilePath)\"\r\n\r\nCompress-Archive -Path \"$($installationPath)\\*\" -DestinationPath $zipFilePath\r\n\r\nWrite-Host \"Deployment zip file created\"\r\n\r\n$username = $OctopusParameters[\"Azf.Username\"]\r\n$password = $OctopusParameters[\"Azf.Password\"]\r\n$appName = $OctopusParameters[\"Azf.ApplicationName\"]\r\n\r\nif(!$username){\r\n    Write-Error \"No Username has been supplied. You can do this from the Step Details page of this step.\"\r\n    \r\n    exit 1;\r\n}\r\n\r\n\r\nif(!$password){\r\n    Write-Error \"No Password has been supplied. You can do this from the Step Details page of this step.\"\r\n    \r\n    exit 1;\r\n}\r\n\r\n\r\nif(!$username){\r\n    Write-Error \"No Application Name has been supplied. You can do this from the Step Details page of this step.\"\r\n    \r\n    exit 1;\r\n}\r\n\r\n$authHeader = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes((\"{0}:{1}\" -f $username,$password)))\r\n\r\n$apiUrl = \"https://$($appName).scm.azurewebsites.net/api/zip/site/wwwroot\"\r\n\r\nWrite-Host \"Uploading deployment zip file to $($apiUrl)\"\r\n\r\nInvoke-RestMethod -Uri $apiUrl -Headers @{Authorization=(\"Basic {0}\" -f $authHeader)} -Method PUT -InFile $zipFilePath -ContentType \"multipart/form-data\"\r\n\r\nWrite-Host \"Upload complete\"\r\n",
+    "Octopus.Action.Package.PackageId": "#{Azf.PackageName}",
+    "Octopus.Action.Package.JsonConfigurationVariablesTargets": "#{Azf.JsonConfigFiles}",
```

Comment:
This is an optional parameter, but if left blank, OD will treat it as the literal `#{Azf.JsonConfigFiles}`. This causes a warning at deployment:

![image](https://user-images.githubusercontent.com/2872984/28026031-cdb1e7fe-658c-11e7-8f11-ba6cb5857565.png)

If anyone can suggest a way to avoid this, I'll update the PR with the changes.